### PR TITLE
Feature: multiple links support

### PR DESCRIPTION
**Pull request by yannbf**
Issue: #5 

This PR adds functionality for multiple links in the addon. It's very useful when a user has multiple layouts per component.
I took the liberty to do a few extra things, feel free to undo them if they do not match your taste:
- Migrated example story to CSF format (default format since storybook 5.3)
- Updated code to use hooks instead of class component
- Added typescript support to internal storybook
- Added stories for the addon components (meta development!)

Also updated README accordingly.

This is just a draft PR, would be great if you could work with me on this! The functionality is there, but UI wise I'd like your help to make it match your vision, therefore I didn't focus on styling :D

ps: would be awesome to have a sample project to use while developing (later on maybe even deploy the storybook so it's easier for users to see the addon working before setting it up).
The links in the README were not working, so I had to use my private project while developing.

### CODE REVIEW

File: src/components/ZeplinPanel.tsx [hunk 1]
```diff
@@ -1,140 +1,142 @@
...
-    fetchZeplinResource = async () => {
-        const { zeplinLink } = this.props;
-
-        if (!zeplinLink) {
+const ZeplinPanel: React.FC<ZeplinPanelProps> = ({ zeplinLink }) => {
+    const [selectedLink, setSelectedLink] = useState("");
+    const [zeplinData, setZeplinData] = useState(null);
+    const [zoomLevel, setZoomLevel] = useState(1);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(null);
```
Comment (mertkahyaoglu) [inline src/components/ZeplinPanel.tsx:21]: Let's combine all of these states into one using `useReducer`. Then we don't need to call consecutive setState functions which causes multiple rerenders.

```js
const [state, setState] = useReducer(
    (currentState: AppState, newState: Partial<AppState>): AppState => ({ ...currentState, ...newState }),
    initialAppState
);
```

Comment (yannbf) [reply src/components/ZeplinPanel.tsx:21]: That sounds like an idea, although if you are concerned regarding multiple rendering, React will batch the setState calls if they're done within a React event handler, same as would happen if you were to call `this.setState({...})` multiple times in a class based component.

https://github.com/facebook/react/issues/14259#issuecomment-439632622

**Comment by yannbf** [comment]
> This is awesome 🎉 Thanks so much @yannbf.
> 
> Just did a quick review. The code is looking good. I'll fully test this when I get a change 👍
> 
> Also, I think it would be better if we could support the old version which `zeplinLink` directly maps to a link. WDYT?

So glad you liked it!
Well, I thought of doing that, and I think it's nice to be retrocompatible, but I wanted to know what was your take in this. We can definitely do that, although it's nice to keep consistency in the API. Let me know what you prefer and we make it work that way!

File: src/components/ZeplinPanel.tsx (hunk 1 above)
Comment (mertkahyaoglu) [reply src/components/ZeplinPanel.tsx:21]: That's good. Thanks for the info 👍. But I'd still prefer using `useReducer` if it works for you too.

Comment (yannbf) [reply src/components/ZeplinPanel.tsx:21]: Definitely!

**Comment by yannbf** [comment]
Changes added as requested! 👯 
I took a different/simpler approach for `useReducer`, although I struggled a bit with the typings for it, so it's not 100% typed. Let me know if that makes sense or if we should indeed create a separate reducer function with action types etc.

File: src/components/ZeplinPanel.tsx [hunk 2]
```diff
@@ -1,140 +1,184 @@
...
+        error,
+    } = state as ZeplinState;
+
+    const fetchZeplinResource = async () => {
+        let designLink: string;
+        if (Array.isArray(zeplinLink)) {
+            designLink = selectedLink || zeplinLink[0].link;
+        } else {
+            designLink = selectedLink || zeplinLink;
+        }
```
Comment (mertkahyaoglu) [inline src/components/ZeplinPanel.tsx:60]:
```suggestion
       designLink = selectedLink || (Array.isArray(zeplinLink) ? zeplinLink[0].link : zeplinLink);
```

File: src/components/ZeplinPanel.tsx [hunk 3]
```diff
@@ -1,140 +1,184 @@
...
+            )}
+        </select>
+    );
+
+    return (
+        <Container>
+            <Header>
+                {LinksSection}
+                <strong>{name}</strong>
+                <i>last updated {lastModifiedAt}</i>
```
Comment (mertkahyaoglu) [inline src/components/ZeplinPanel.tsx:155]:
```suggestion
                <i>Last updated: {lastModifiedAt}</i>
```

**Comment by yannbf** [comment]
Great review. thanks!! All of them have been addressed.

**Comment by mertkahyaoglu** [comment]
@yannbf Great work! There are a few things left. I'm going to create a sample Zeplin project. So we can use it for multiple links in the storybook example and also for readme links 👍

**Comment by yannbf** [comment]
Awesome! Feel free to push changes in my branch, as I mentioned I didn't do anything regarding styles so far.

> I'm going to create a sample Zeplin project. So we can use it for multiple links in the storybook example and also for readme links

Great. It makes it easier for anyone else to contribute to this repo as well! 🚀

**Comment by yannbf** [comment]
Just checked the changes and ran the project, it's all 👌 
And just so you know, I tried to run with the links from the codebase but it says **Project not found**, then when opening the link directly, Zeplin says **You’re not invited to this project, Please contact project members.**

**Comment by mertkahyaoglu** [comment]
@yannbf I created a sample project and deployed to [Netlify](https://storybook-zeplin.netlify.app). I invited you to Zeplin project with the email in your bio.

I had to rebase this branch and made a few changes regarding to story examples and readme. I also deleted `ZeplinPanel.stories.jsx` if you don't mind. Because ZeplinPanel is already visible in the panel via addon 😄 I didn't touch the design for now. We can make it pretty later. AFAIS, this is looking good and ready to go 🎉 WDYT?

**Comment by yannbf** [comment]
Thanks a lot @mertkahyaoglu! I understand the reason you removed the stories, no worries at all! 
I think it's great and a result of a nice collaboration <3
As soon as you ship this it will unblock me in my company project so I will be very thankful for that!

You can count on me to check on more features for this addon!

**Comment by mertkahyaoglu** [comment]
Thanks @yannbf ❤️ This is a really good feature and I enjoyed working with you. I'm adding you as a collaborator. Feel free to come up with your ideas anytime.

I'm publishing a new version for this feature now 👍

**Pull request merged**